astroquery/mast/core.py

```diff
@@ -944,8 +960,11 @@ def query_criteria_async(self, pagesize=None, page=None, **criteria):
         # grabbing the observation type (science vs calibration)
         obstype = criteria.pop('obstype', 'science')
 
-        # Build the mashup filter object
-        mashupFilters = self._build_filter_set(**criteria)
+        # Build the mashup filter object and store it in the correct service_name entry
+        if coordinates or objectname:
```

Comment:
if `coordinates` is specified, will it always evaluate as `True`?  Is there a sensible way in which this could evaluate as  `False`?

Reply:
If neither coordinates or objectname are specified it will evaluate false. 

Basically if either objectname or coordinates are specified we are doing a filtered query around a specific position, while if neither of them are specified we are doing a non-positional filtered query, and those require difference services on the MAST server side.